src/server/coder/workspaces.ts

```diff
@@ -131,3 +143,26 @@ export async function createUserAndWorkspace(
         )
     }
 }
+
+const initializeWorkspaceCodeFiles = async (studyId: string): Promise<void> => {
+    const coderBaseFilePath = await getConfigValue('CODER_FILES')
+    const latestJob = await latestJobForStudy(studyId)
+    const latestJobInfo = await jobInfoForJobId(latestJob.id)
+
+    const mainCodeFile = latestJob.files.filter((file) => file.fileType === 'MAIN-CODE')
+    const supplementalCodeFiles = latestJob.files.filter((file) => file.fileType === 'SUPPLEMENTAL-CODE')
+
+    for (const codeFile of mainCodeFile.concat(supplementalCodeFiles)) {
+        const fileData = await fetchFileContents(
+            pathForStudyJobCodeFile(
+                { orgSlug: latestJobInfo.orgSlug, studyId: latestJob.studyId, studyJobId: latestJob.id },
+                codeFile.name,
+            ),
+        )
+        const targetFilePath = path.join(coderBaseFilePath, studyId, codeFile.name)
+
+        // Create parent directory if needed and then write file
+        await fs.mkdir(path.dirname(targetFilePath), { recursive: true })
+        await fs.writeFile(targetFilePath, Buffer.from(await fileData.arrayBuffer()))
+    }
+}
```

Comment:
this is fine for now, but don't spend a huge amount of time on it, we've got new designs starting next week to support users uploading code directly.